collections/_article/rendering-acyclic-graph.md

```diff
@@ -0,0 +1,279 @@
+
+#### Barriers
+
+The biggest change compared to previous APIs is that synchronization between commands is no longer automatically deduced by the driver and it must be manually implemented by the programmers using Vulkan. If you want to read more about this topic in detail, I highly recommend Hans-Kristian's [blog post about Vulkan Synchronization](https://themaister.net/blog/2019/08/14/yet-another-blog-explaining-vulkan-synchronization/) and reading the [official Vulkan documentation](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdPipelineBarrier.html). To keep the article short, a very basic explanation is provided below.
+
+The order of execution of the recorded commands inside a command buffer **is NOT guaranteed to complete in the order they were submitted in**: the GPU can reorder these commands in whatever order it thinks is best to complete the job as quickly as possible. To compensate for this, the programmer must manually insert **[synchronization barriers](https://gpuopen.com/learn/vulkan-barriers-explained/)** in the command buffer that allow specifying in detail which commands should be completed or start by specifying the scope **before** and **after**. The scope includes multiple concepts such as the execution stage (e.g. drawing, compute, transfer), the type of access (e.g. read or write) and the affected memory regions (globally, buffers or textures). On top of that, barriers are capable of transitioning textures from one memory layout to another, which is a requirement to be able to use textures in different commands optimally.
+
+| ![](/storage/blog/acyclic-graph/barriers.webp) |
+|:--:| 
+| *A barrier can establish a dependency between commands by specifying a synchronization scope.* |
+
+Beginners will have a hard time understanding this mechanism, but even experts or IHVs are not safe from making mistakes. A run of the [Vulkan Validation Layers](https://github.com/KhronosGroup/Vulkan-ValidationLayers) in *synchronization* mode will reveal multiple issues on most Godot projects or even commercial games available in the market. These are among the most frustrating issues to understand as they won't even appear consistently depending on hardware vendor or hardware speed. Since eliminating these problems was one of the main goals behind the introduction of the acyclic graph, extensive use was made of these validation tools to ensure that no synchronization errors remained.
```

Comment:
```suggestion
Beginners will have a hard time understanding this mechanism, but even experts or hardware vendors are not safe from making mistakes. A run of the [Vulkan Validation Layers](https://github.com/KhronosGroup/Vulkan-ValidationLayers) in *synchronization* mode will reveal multiple issues on most Godot projects or even commercial games available in the market. These are among the most frustrating issues to understand as they won't even appear consistently depending on hardware vendor or hardware speed. Since eliminating these problems was one of the main goals behind the introduction of the acyclic graph, extensive use was made of these validation tools to ensure that no synchronization errors remained.
```